src/wp-includes/option.php

```diff
@@ -922,6 +922,15 @@ function update_option( $option, $value, $autoload = null ) {
 	 */
 	do_action( 'update_option', $option, $old_value, $value );
 
+	/*
+	 * Ensure the serialized value is a string.
+	 *
+	 * This ensure that the option is stored in the cache in the same format as the
+	 * option is stored in the database. Rather than type casting, sprintf is used to
+	 * match the process used by wpdb::prepare().
+	 */
+	$serialized_value = sprintf( '%s', $serialized_value );
```

Comment:
As far as I can tell, the `$serialized_value` variable isn't used anywhere between lines 905 and 923. Could we move this right below the clause in 905 to better colocated the code that adjusts the option value to store?